### PR TITLE
WT-12731 Add ex_file_system to examples-c-tsan

### PR DESCRIPTION
**Pull request by ajmorton**
Add ex_file_system to examples-c-tsan and resolve any new TSan warnings

### CODE REVIEW

File: src/evict/evict_lru.c [hunk 1]
```diff
@@ -1569,7 +1569,8 @@ __evict_walk(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue)
         /*
          * If we are filling the queue, skip files that haven't been useful in the past.
          */
-        if (btree->evict_walk_period != 0 && btree->evict_walk_skips++ < btree->evict_walk_period) {
+        if (__wt_atomic_load32(&btree->evict_walk_period) != 0 &&
```
Comment (luke-pearson) [inline src/evict/evict_lru.c:1572]: Should this be pulled into a local variable?

File: src/evict/evict_lru.c [hunk 2]
```diff
@@ -2211,16 +2213,18 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
...
-        btree->evict_walk_period = 0;
+        __wt_atomic_store32(&btree->evict_walk_period, 0);
         /*
          * If there's a chance the Btree was fully evicted, update the evicted flag in the handle.
          */
         if (__wt_btree_bytes_evictable(session) == 0)
             FLD_SET(session->dhandle->advisory_flags, WT_DHANDLE_ADVISORY_EVICTED);
-    } else if (btree->evict_walk_period > 0)
-        btree->evict_walk_period /= 2;
+    } else if (__wt_atomic_load32(&btree->evict_walk_period) > 0)
```
Comment (luke-pearson) [inline src/evict/evict_lru.c:2225]: Another case for a local variable.

File: src/log/log_slot.c [hunk 3]
```diff
@@ -629,9 +630,9 @@ __wti_log_slot_join(WT_SESSION_IMPL *session, uint64_t mysize, uint32_t flags, W
     if (LF_ISSET(WT_LOG_FSYNC))
         F_SET_ATOMIC_16(slot, WT_SLOT_SYNC);
     if (F_ISSET(myslot, WT_MYSLOT_UNBUFFERED)) {
-        WT_ASSERT(session, slot->slot_unbuffered == 0);
+        WT_ASSERT(session, __wt_atomic_loadi64(&slot->slot_unbuffered) == 0);
         WT_STAT_CONN_INCR(session, log_slot_unbuffered);
-        slot->slot_unbuffered = (int64_t)mysize;
+        __wt_atomic_storei64(&slot->slot_unbuffered, (int64_t)mysize);
```
Comment (luke-pearson) [inline src/log/log_slot.c:635]: So atomic storing a uint into an int. Weird but that's already the code.

File: src/evict/evict_lru.c (hunk 1 above)
Comment (ajmorton) [reply src/evict/evict_lru.c]: Updated in [c088090](https://github.com/wiredtiger/wiredtiger/pull/10635/commits/c08809048d3122fa26da500e7ecda927e373e41d)

File: src/evict/evict_lru.c (hunk 2 above)
Comment (ajmorton) [reply src/evict/evict_lru.c]: Done in [c088090](https://github.com/wiredtiger/wiredtiger/pull/10635/commits/c08809048d3122fa26da500e7ecda927e373e41d)

**Pull request merged**